### PR TITLE
python3Packages.sortedcollections: 1.2.1 -> 1.2.3

### PR DESCRIPTION
**Pull request by fabaff**
###### Motivation for this change
Update to latest upstream release 1.2.3

Commit log: https://github.com/grantjenks/python-sortedcollections/commits/master

`attrs` is at 20.3.0 and this leads to build failures. 

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/google-music-proto/default.nix [hunk 1]
```diff
@@ -20,7 +20,8 @@ buildPythonPackage rec {
   postPatch = ''
     sed -i -e "/audio-metadata/c\'audio-metadata'," -e "/marshmallow/c\'marshmallow'," setup.py
     substituteInPlace setup.py \
-      --replace "pendulum>=2.0,<=3.0,!=2.0.5,!=2.1.0" "pendulum>=2.0,<=3.0"
+      --replace "pendulum>=2.0,<=3.0,!=2.0.5,!=2.1.0" "pendulum>=2.0,<=3.0" \
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/google-music-proto/default.nix:23]: Did you check if this still applies?

File: pkgs/development/python-modules/audio-metadata/default.nix [hunk 2]
```diff
@@ -20,7 +20,8 @@ buildPythonPackage rec {
     substituteInPlace setup.py \
       --replace "bidict>=0.17,<0.18" "bidict" \
       --replace "more-itertools>=4.0,<8.0" "more-itertools" \
-      --replace "pendulum>=2.0,<=3.0,!=2.0.5,!=2.1.0" "pendulum>=2.0,<=3.0"
+      --replace "pendulum>=2.0,<=3.0,!=2.0.5,!=2.1.0" "pendulum>=2.0,<=3.0" \
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/audio-metadata/default.nix:23]: Did you check if this and the above still apply?

File: pkgs/development/python-modules/tbm-utils/default.nix [hunk 3]
```diff
@@ -22,9 +22,11 @@ buildPythonPackage rec {
 
   # this versioning was done to prevent normal pip users from encountering
   # issues with package failing to build from source, but nixpkgs is better
+  # attrs: https://github.com/thebigmunch/tbm-utils/issues/1
   postPatch = ''
     substituteInPlace setup.py \
-      --replace "pendulum>=2.0,<=3.0,!=2.0.5,!=2.1.0" "pendulum>=2.0,<=3.0"
+      --replace "pendulum>=2.0,<=3.0,!=2.0.5,!=2.1.0" "pendulum>=2.0,<=3.0" \
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/tbm-utils/default.nix:28]: Did you check if this still applies?

File: pkgs/tools/audio/google-music-scripts/default.nix [hunk 4]
```diff
@@ -30,7 +30,8 @@ buildPythonApplication rec {
   postPatch = ''
     substituteInPlace setup.py \
       --replace "tomlkit>=0.5,<0.6" "tomlkit" \
-      --replace "pendulum>=2.0,<=3.0,!=2.0.5,!=2.1.0" "pendulum"
+      --replace "pendulum>=2.0,<=3.0,!=2.0.5,!=2.1.0" "pendulum" \
```
Comment (SuperSandro2000) [inline pkgs/tools/audio/google-music-scripts/default.nix:33]: Did you check if this still applies?

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 108572` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package marked as broken and skipped:
</summary>
<ul>
<li>python39Packages.google-music</li>
</ul>
</details>
<details>

  <summary>2 packages failed to build and are new build failures:</summary>
  <ul>
<li>python39Packages.flask-socketio: log was empty</li>
<li>python39Packages.python-socketio: log was empty</li>
  </ul>
</details>
<details>
<summary>
27 packages built:
</summary>
<ul>
<li>appdaemon</li>
<li>gdbgui</li>
<li>google-music-scripts</li>
<li>python37Packages.audio-metadata</li>
<li>python37Packages.bidict</li>
<li>python37Packages.flask-socketio</li>
<li>python37Packages.google-music</li>
<li>python37Packages.google-music-proto</li>
<li>python37Packages.google-music-utils</li>
<li>python37Packages.python-socketio</li>
<li>python37Packages.sortedcollections</li>
<li>python37Packages.tbm-utils</li>
<li>python38Packages.audio-metadata</li>
<li>python38Packages.bidict</li>
<li>python38Packages.flask-socketio</li>
<li>python38Packages.google-music</li>
<li>python38Packages.google-music-proto</li>
<li>python38Packages.google-music-utils</li>
<li>python38Packages.python-socketio</li>
<li>python38Packages.sortedcollections</li>
<li>python38Packages.tbm-utils</li>
<li>python39Packages.audio-metadata</li>
<li>python39Packages.bidict</li>
<li>python39Packages.google-music-proto</li>
<li>python39Packages.google-music-utils</li>
<li>python39Packages.sortedcollections</li>
<li>python39Packages.tbm-utils</li>
</ul>
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 108572` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
29 packages built:
</summary>
<ul>
<li>appdaemon</li>
<li>gdbgui</li>
<li>google-music-scripts</li>
<li>python37Packages.audio-metadata</li>
<li>python37Packages.bidict</li>
<li>python37Packages.flask-socketio</li>
<li>python37Packages.google-music</li>
<li>python37Packages.google-music-proto</li>
<li>python37Packages.google-music-utils</li>
<li>python37Packages.python-socketio</li>
<li>python37Packages.sortedcollections</li>
<li>python37Packages.tbm-utils</li>
<li>python38Packages.audio-metadata</li>
<li>python38Packages.bidict</li>
<li>python38Packages.flask-socketio</li>
<li>python38Packages.google-music</li>
<li>python38Packages.google-music-proto</li>
<li>python38Packages.google-music-utils</li>
<li>python38Packages.python-socketio</li>
<li>python38Packages.sortedcollections</li>
<li>python38Packages.tbm-utils</li>
<li>python39Packages.audio-metadata</li>
<li>python39Packages.bidict</li>
<li>python39Packages.flask-socketio</li>
<li>python39Packages.google-music-proto</li>
<li>python39Packages.google-music-utils</li>
<li>python39Packages.python-socketio</li>
<li>python39Packages.sortedcollections</li>
<li>python39Packages.tbm-utils</li>
</ul>
</details>

**Comment by fabaff** [comment]
#108536 was addressing a couple of the same build issues.

File: pkgs/tools/audio/google-music-scripts/default.nix (hunk 4 above)
Comment (fabaff) [reply pkgs/tools/audio/google-music-scripts/default.nix]: Updated

File: pkgs/development/python-modules/tbm-utils/default.nix (hunk 3 above)
Comment (fabaff) [reply pkgs/development/python-modules/tbm-utils/default.nix]: No longer relevant.

File: pkgs/development/python-modules/audio-metadata/default.nix (hunk 2 above)
Comment (fabaff) [reply pkgs/development/python-modules/audio-metadata/default.nix]: No longer relevant.

File: pkgs/development/python-modules/google-music-proto/default.nix (hunk 1 above)
Comment (fabaff) [reply pkgs/development/python-modules/google-music-proto/default.nix]: No longer relevant.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 108572` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package marked as broken and skipped:
</summary>
<ul>
<li>python39Packages.google-music</li>
</ul>
</details>
<details>
<summary>
26 packages built:
</summary>
<ul>
<li>appdaemon</li>
<li>gdbgui</li>
<li>google-music-scripts</li>
<li>python37Packages.audio-metadata</li>
<li>python37Packages.bidict</li>
<li>python37Packages.flask-socketio</li>
<li>python37Packages.google-music</li>
<li>python37Packages.google-music-proto</li>
<li>python37Packages.google-music-utils</li>
<li>python37Packages.python-socketio</li>
<li>python37Packages.sortedcollections</li>
<li>python38Packages.audio-metadata</li>
<li>python38Packages.bidict</li>
<li>python38Packages.flask-socketio</li>
<li>python38Packages.google-music</li>
<li>python38Packages.google-music-proto</li>
<li>python38Packages.google-music-utils</li>
<li>python38Packages.python-socketio</li>
<li>python38Packages.sortedcollections</li>
<li>python39Packages.audio-metadata</li>
<li>python39Packages.bidict</li>
<li>python39Packages.flask-socketio</li>
<li>python39Packages.google-music-proto</li>
<li>python39Packages.google-music-utils</li>
<li>python39Packages.python-socketio</li>
<li>python39Packages.sortedcollections</li>
</ul>
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 108572` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
25 packages built:
</summary>
<ul>
<li>appdaemon</li>
<li>gdbgui</li>
<li>google-music-scripts</li>
<li>python37Packages.audio-metadata</li>
<li>python37Packages.bidict</li>
<li>python37Packages.flask-socketio</li>
<li>python37Packages.google-music</li>
<li>python37Packages.google-music-proto</li>
<li>python37Packages.google-music-utils</li>
<li>python37Packages.python-socketio</li>
<li>python37Packages.sortedcollections</li>
<li>python38Packages.audio-metadata</li>
<li>python38Packages.bidict</li>
<li>python38Packages.flask-socketio</li>
<li>python38Packages.google-music</li>
<li>python38Packages.google-music-proto</li>
<li>python38Packages.google-music-utils</li>
<li>python38Packages.python-socketio</li>
<li>python38Packages.sortedcollections</li>
<li>python39Packages.audio-metadata</li>
<li>python39Packages.bidict</li>
<li>python39Packages.flask-socketio</li>
<li>python39Packages.google-music-proto</li>
<li>python39Packages.python-socketio</li>
<li>python39Packages.sortedcollections</li>
</ul>
</details>

**Pull request merged**